drivers/led/leds_gpio.c

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright (c) 2018 Manivannan Sadhasivam
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief LEDs driver for GPIOs
+ *
+ * This driver controls LED devices attached to GPIO pins which
+ * are declared under leds node in board devicetree. The driver
+ * takes care of configuring the pin to output state before using it.
+ *
+ * TODO: Implement Software Blinking
+ */
+
+#include <board.h>
+#include <device.h>
+#include <gpio.h>
+#include <led.h>
+#include <misc/util.h>
+#include <zephyr.h>
+
+#define SYS_LOG_LEVEL CONFIG_SYS_LOG_LED_LEVEL
+#include <logging/sys_log.h>
+
+#include "led_context.h"
+
+enum led_gpio_modes {
+	LED_OFF,
+	LED_ON,
+};
+
+struct led_gpio_cfg {
+	char *gpio_port;
+	u8_t gpio_pin;
+	u8_t gpio_polarity;
+};
+
+struct led_gpio_data {
+	struct device *gpio;
+	struct led_data dev_data;
+};
+
+static inline int led_gpio_on(struct device *dev, u32_t led)
+{
+	const struct led_gpio_cfg *cfg = dev->config->config_info;
+	struct led_gpio_data *data = dev->driver_data;
+
+	/* Make sure that the LED is valid */
+	if (led != cfg->gpio_pin)
+		return -EINVAL;
+
+	/*
+	 * Check for active high or active low defined in DT. For
+	 * simplicity, non zero cases are considered as active high
+	 * and zeroes are active low.
+	 */
+	if (cfg->gpio_polarity)
+		gpio_pin_write(data->gpio, led, LED_ON);
+	else
+		gpio_pin_write(data->gpio, led, LED_OFF);
```

Comment:
How about replacing the if/else with:
```c
gpio_pin_write(data->gpio, led, (cfg->gpio_polarity & GPIO_INT_ACTIVE_HIGH) != 0);
```

Reply:
yeah, but I just wanted to make it explicit for the user here. Anyway, I can modify this.